src/Chapter2.hs

```diff
@@ -842,7 +856,10 @@ list.
 
 🕯 HINT: Use the 'cycle' function
 -}
-rotate = error "rotate: Not implemented!"
+rotate :: Int -> [a] -> [a]
+rotate n list
+  | n < 0 = []
+  | otherwise = (take (length list)) (drop n (cycle list))
```

Comment:
That's a correct solution! Though, it does a little bit more work when the given number `n` is big and the list is small. You will cycle through the list a lot. Maybe you can think of a way to optimize this :slightly_smiling_face:

Reply:
Here our friend `mod` can help!